### ISSUE TITLE
should report unused_import when it reexports nothing

### ISSUE THREAD

**Issue by bvanjoi** [rust-lang/rust#116032]
### Code

```Rust
mod a {}

pub use a::*;

fn main() {}
```

### Current output

```Shell
nothing
```

### Desired output

```Shell
unused import: `a::*`
```
